web/src/components/data-fields/address/AddressTextField.tsx

```diff
@@ -0,0 +1,69 @@
+import { GenericTextField, GenericTextFieldProps } from "@/components/GenericTextField";
+import { ModifiedContent } from "@/components/ModifiedContent";
+import { WithErrorBar } from "@/components/WithErrorBar";
+import { AddressContext } from "@/contexts/addressContext";
+import { ProfileFormContext } from "@/contexts/profileFormContext";
+import { useAddressErrors } from "@/lib/data-hooks/useAddressErrors";
+import { useFormContextFieldHelpers } from "@/lib/data-hooks/useFormContextFieldHelpers";
+import { Address } from "@businessnjgovnavigator/shared/";
+import { ReactElement, useContext } from "react";
+
+export interface Props extends Omit<GenericTextFieldProps, "value" | "error" | "inputWidth"> {
+  fieldName: keyof Address;
+  label?: string;
+  secondaryLabel?: string;
+  errorBarType: "ALWAYS" | "MOBILE-ONLY" | "DESKTOP-ONLY" | "NEVER";
+}
+
+export const AddressTextField = ({ className, ...props }: Props): ReactElement => {
+  const { state, setAddressData } = useContext(AddressContext);
+  const { doesFieldHaveError } = useAddressErrors();
+  const { setIsValid } = useFormContextFieldHelpers(props.fieldName, ProfileFormContext);
+  const { setIsValid: setIsValidAddressLine1 } = useFormContextFieldHelpers(
+    "addressLine1",
+    ProfileFormContext
+  );
+  const { setIsValid: setIsValidMunicipality } = useFormContextFieldHelpers(
+    "addressMunicipality",
+    ProfileFormContext
+  );
+  const { setIsValid: setIsValidZipCode } = useFormContextFieldHelpers("addressZipCode", ProfileFormContext);
+
+  const handleChange = (value: string): void => {
+    props.handleChange && props.handleChange(value);
+    setAddressData((prevAddressData) => ({ ...prevAddressData, [props.fieldName]: value }));
+  };
+
+  const onValidation = (): void => {
+    const partialAddressValidation = new Set(["addressLine1", "addressZipCode"]);
+
+    if (partialAddressValidation.has(props.fieldName)) {
+      setIsValidAddressLine1(!doesFieldHaveError("addressLine1"));
+      setIsValidMunicipality(!doesFieldHaveError("addressMunicipality"));
+      setIsValidZipCode(!doesFieldHaveError("addressZipCode"));
+    } else {
+      setIsValid(!doesFieldHaveError(props.fieldName));
+    }
+  };
```

Comment:
[pebble] Given we know we're going to have 2 other instances where we'll be adding address to profile, and they'll likely be using the same components to do this, I think it makes sense to move validation to the parent `ProfileAddressFields` (to be renamed `ProfileNJAddress`) component. This will just make it easier to implement the following address components on the profile for Poppy and Dakota.

Additionally, move validation tests to parent level as well.